api/processors/services/group_builder.py

```diff
@@ -282,3 +276,37 @@ def _create_log_events(group: Group) -> None:
             teacher_log_event_type=TeacherLogEventType.GROUP_OFFERED,
             group_log_event_type=GroupLogEventType.FORMED,
         )
+
+    @staticmethod
+    def _compare_groups_priority(group1: GroupCandidate, group2: GroupCandidate) -> int:
+        """
+        Compare priority of two group candidates.
+        Returns:
+            -1 if group1 has higher priority
+            0 if both candidates have equal priority
+            1 if group2 has higher priority
+        """
+        level_difference = abs(
+            group1.language_and_level.level.index - group2.language_and_level.level.index
+        )
+        time_threshold = timezone.now() - MAX_WAITING_TIME
+
+        if level_difference >= LEVEL_DIFFERENCE_THRESHOLD:
+            # If levels difference is 2 or more, prefer the lower level
+            if group1.language_and_level.level.index < group2.language_and_level.level.index:
+                return -1
+            return 1
```

Comment:
hm... not sure I understand why we're already returning `1` here without further checks

Reply:
Good catch. Reworked it - now we return 1 only if index1 is more than index2, but if they are same - we do compare by waiting time